packages/rpc-core/src/bundle.ts

```diff
@@ -85,6 +85,7 @@ export class RpcCore {
   #registryDefault: Registry;
 
   #getBlockRegistry?: (blockHash: Uint8Array) => Promise<{ registry: Registry }>;
+  #getBlockHash?: (blockNumber?: BlockNumber | AnyNumber | undefined) => Promise<Uint8Array | string | null | undefined>;
```

Comment:
```suggestion
  #getBlockHash?: (blockNumber: AnyNumber) => Promise<Uint8Array>;
```

`AnyNumber` will cater for it. Additionally this should always return a value. (It not, it will throw)